### PR TITLE
Raise ``NotImplementedError`` for unsupported ``SeriesGroupBy.var/std``

### PR DESCRIPTION
**Pull request by rjzamora**
Closes https://github.com/dask-contrib/dask-expr/issues/840

This change is not entirely necessary given that the "correct" result is trivial. However, I think it's best to provide a bit of information to the user since it's easy.

### CODE REVIEW

File: dask_expr/_groupby.py [hunk 1]
```diff
@@ -1743,6 +1743,11 @@ def var(
             or is_scalar(self._slice)
             and self._slice is not None
         ):
+            if len(result.columns) < 1:
+                raise NotImplementedError(
+                    "Cannnot call `SeriesGroupBy.var` on the key column. "
```
Comment (hendrikmakait) [inline dask_expr/_groupby.py:1748]:
```suggestion
                    "Cannot call `SeriesGroupBy.var` on the key column. "
```

File: dask_expr/_groupby.py [hunk 2]
```diff
@@ -1778,6 +1783,11 @@ def std(
             or is_scalar(self._slice)
             and self._slice is not None
         ):
+            if len(result.columns) < 1:
+                raise NotImplementedError(
+                    "Cannnot call `SeriesGroupBy.std` on the key column. "
```
Comment (hendrikmakait) [inline dask_expr/_groupby.py:1788]:
```suggestion
                    "Cannot call `SeriesGroupBy.std` on the key column. "
```

File: dask_expr/_groupby.py (hunk 1 above)
Comment (rjzamora) [reply dask_expr/_groupby.py:1748]: Wow. Thanks for catching that!

**Comment by phofl** [comment]
thx

**Pull request merged**